osu.Desktop/DiscordRichPresence.cs

```diff
@@ -96,6 +98,21 @@ private void updateStatus()
             client.SetPresence(presence);
         }
 
+        private string truncate(ReadOnlySpan<char> str)
+        {
+            if (Encoding.UTF8.GetByteCount(str) <= 128)
+                return new string(str);
+
+            int ellipsisLength = Encoding.UTF8.GetByteCount(new[] { '…' });
+
+            do
+            {
+                str = str[..^1];
+            } while (Encoding.UTF8.GetByteCount(str) + ellipsisLength > 128);
+
+            return new string(str) + '…';
```

Comment:
Allocation here can be reduced:
assuming `ReadOnlyMemory<char> str`:
```csharp
return string.Create(str.Length + 1, str, (span, s) =>
{
    s.Span.Copyto(span);
    span[^1] = '…';
});
```

Reply:
It's good to know, but I'm not interested in micro-optimisations like this in a low-overhead component in the first place. See other comment also - this also gets kinda ugly due to `ReadOnlyMemory<>` shortcomings.